src/main/java/uk/ac/cam/cl/dtg/segue/api/UsersFacade.java

```diff
@@ -852,4 +826,65 @@ private Response createUserObjectAndLogIn(final HttpServletRequest request, fina
                     "You cannot register with an Isaac email address.").toResponse();
         }
     }
+
+
+    /**
+     * Convert user-provided preference maps to UserPreference lists.
+     *     Contains Isaac classes which ideally should not be here . . .
+     *
+     * @param userPreferenceObject
+     *            - the user-provided preference object
+     * @param userId
+     *            - the userId of the user
+     * @return whether the preference is valid
+     */
+    private List<UserPreference> userPreferenceObjectToList(final Map<String, Map<String, Boolean>> userPreferenceObject, final long userId) {
+        List<UserPreference> userPreferences = Lists.newArrayList();
+        if (null == userPreferenceObject) {
+            return userPreferences;
+        }
+        // FIXME: This entire method is horrible, but required to sanitise what is stored in the database . . .
+        for (String preferenceType: userPreferenceObject.keySet()) {
+
+            // Check if the given preference type is one we support:
+            if (!EnumUtils.isValidEnum(IsaacUserPreferences.class, preferenceType)
```

Comment:
We could make this slightly nicer by dependency injecting all the enums we should support into a single list. This will also remove the need for specifying isaac specific classes in the code. This is fine for now though.